api/generator/generator.go

```diff
@@ -266,22 +266,20 @@ func (cg *CertGeneratorHandler) Handle(w http.ResponseWriter, r *http.Request) e
 			"certificate":         certSum,
 		},
 	}
+
+	if len(req.Request.Hosts) == 0 {
+		return api.SendResponseWithMessage(w, result,
+			"This certificate lacks a \"hosts\" field. This makes it unsuitable for websites. " +
```

Comment:
For clarity, this could be a constant at the beginning of the file. You could also do the following:

```
var csrNoHostMessage = `This certificate lacks a "hosts" field. This makes it unsuitable for websites. For more information see the Baseline Requirements for the Issuance and Management of Publicly-Trusted Certificates, v.1.1.6, from the CA/Browser Forum (https://cabforum.org); specifically, section 10.2.3 ("Information Requirements").`
```